workflow/snakemake_rules/nextclade.smk

```diff
@@ -179,18 +180,19 @@ rule run_nextclade:
     params:
         genes=GENES_SPACE_DELIMITED,
         translation_arg=lambda w: (
-            f"--output-translations=data/{database}/nextclade{w.reference}.translation_{{gene}}.upd.fasta"
+            # Nextclade takes a filename template in which it replaces {gene}
+            # itself, so we want to pass thru {gene} literally to it and make
+            # sure it isn't interpretted by the shell as a glob.  We shellquote
+            # here instead of with :q below because we don't want to pass an
+            # empty string argument when this param is empty.
+            shellquote(f"--output-translations=data/{database}/nextclade{w.reference}.translation_{{gene}}.upd.fasta")
             if w.reference == ""
             else ""
         ),
-        translations=lambda w: [
-            f"data/{database}/nextclade{w.reference}.translation_{gene}.upd.fasta"
-            for gene in GENE_LIST
-        ],
     output:
         info=f"data/{database}/nextclade{{reference}}_new_raw.tsv",
-        alignment=f"data/{database}/nextclade{{reference}}.aligned.upd.fasta",
-        touchfile=touch(f"data/{database}/nextclade{{reference}}.touch"),
+        alignment=temp(f"data/{database}/nextclade{{reference}}.aligned.upd.fasta"),
+        translations=[temp(f"data/{database}/nextclade{{reference}}.translation_{gene}.upd.fasta") for gene in GENE_LIST],
```

Comment:
There may be an issue here during a non-dry run because the builds where `reference == "_21L"` won't pass `--output-translations` and so won't produce these output files. I think Snakemake is likely to notice and error, but I don't know it's exact behaviour off hand. The simplest solution, since we can't condition the outputs on a wildcard, may be to remove the condition on `--output-translations` and always produce them, even if they're immediately removed because no downstream rules use them. Alternative solutions include producing them instead in the `shell:` block with `touch` when necessary for `reference != ""`.

Reply:
> even if they're immediately removed because no downstream rules use them

Oh! I didn't realize that Snakemake removes temp files immediately if no downstream rules use them!

Reply:
Ok now I see how differentiating between main and 21L as suggested by @joverlee521 above makes things simpler :)